### DISCUSSION TITLE
Cannot build EmulatorPkg

### DISCUSSION THREAD

**Discussion by LudekZaloudek** [tianocore/edk2#4286] [Q&A]
Hi. I have tried building EmulatorPkg on Win 10 based on the guide here: https://github.com/tianocore/tianocore.github.io/wiki/EmulatorPkg

However, I am quite disappointed because the guide is obviously missing some setting as I get the following wile invoking Stuart like this:

`stuart_build -c EmulatorPkg/PlatformCI/PlatformBuild.py -t VS2019 -a X64
`
Ater running for some time, I get the following issue:

```
ERROR - Traceback (most recent call last):
  File "C:\EDKII\edk2\.venv\lib\site-packages\edk2toolext\environment\uefi_build.py", line 145, in Go
    ret = self.SetEnv()
  File "C:\EDKII\edk2\.venv\lib\site-packages\edk2toolext\environment\uefi_build.py", line 478, in SetEnv
    "BUILD_OUT_TEMP"), target + "_" + self.env.GetValue("TOOL_CHAIN_TAG")), "Computed in SetEnv")
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```
I have tried setting the BUILD_OUT_TEMP and TOOL_CHAIN_TAG env variables and also putting them into the Conf/Target.txt to no avail. Can someone please point me in the right direction?

**Comment by spbrogan** [comment]
Have you been following the platform specific directions found here:
https://github.com/tianocore/edk2/blob/master/EmulatorPkg/PlatformCI/ReadMe.md

Your build command looks off.  
stuart_build -c EmulatorPkg/PlatformCI/PlatformBuild.py TOOL_CHAIN_TAG=<TOOL_CHAIN_TAG> -a <TARGET_ARCH>

**Comment by ghost** [comment]
For the stuart tools, `-t` is now the target name - e.g. `-t DEBUG`.

**Comment by makubacki** [comment]
To confirm, I tried building against the latest edk2 `master` branch (`6fb2760`) and could do so with the following commands:

- `stuart_update -c EmulatorPkg/PlatformCI/PlatformBuild.py TOOL_CHAIN_TAG=VS2019 -a X64`
- `stuart_build -c EmulatorPkg/PlatformCI/PlatformBuild.py TOOL_CHAIN_TAG=VS2019 -a X64`

Please make sure you've also set up a Python virtual environment and installed the PIP modules in `pip-requirements.txt`.

**Comment by LudekZaloudek** [comment]
Hello, thanks for the tips, I will try these when I get back to it. Had to move on to other tasks in the meantime.


Luděk Žaloudek
Systems Engineering
NXP Semiconductors
Sochorova 3232/36
616 00 Brno - Žabovřesky
Czech Republic
***@***.******@***.***>

***@***.***

From: Michael Kubacki ***@***.***>
Sent: středa 10. května 2023 15:54
To: tianocore/edk2 ***@***.***>
Cc: Ludek Zaloudek ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [tianocore/edk2] Cannot build EmulatorPkg (Discussion #4286)

Caution: This is an external email. Please take care when clicking links or opening attachments. When in doubt, report the message using the 'Report this email' button


To confirm, I tried building against the latest edk2 master branch (6fb2760) and could do so with the following commands:

  *   stuart_update -c EmulatorPkg/PlatformCI/PlatformBuild.py TOOL_CHAIN_TAG=VS2019 -a X64
  *   stuart_build -c EmulatorPkg/PlatformCI/PlatformBuild.py TOOL_CHAIN_TAG=VS2019 -a X64

Please make sure you've also set up a Python virtual environment and installed the PIP modules in pip-requirements.txt.

-

**Comment by Perry31** [comment]
Hello All,
 I'm trying to build EmulatorPkg in Windows environment. But I couldn't make it successful.
 
 Below are the details i'm using,
 - UDK2017
 - I've followed the instruction given in 
 https://github.com/tianocore/edk2/blob/master/EmulatorPkg/PlatformCI/ReadMe.md

![image](https://github.com/tianocore/edk2/assets/13589548/c9f9a040-6002-4ac3-a698-baee6fd380cf)

I'm able to build Nt32Pkg and successfully launched Nt32 simulator. 
But not able to build EmulatorPkg.

Kindly guide me to build EmulatorPkg.

I'm quite curious to know the difference between Nt32Pkg and EmulatorPkg and its advantages.
